src/midend/mir.cpp

```diff
@@ -181,6 +181,25 @@ namespace graphit {
             return node;
         }
 
+
+        void ConstantVectorExpr::copy(MIRNode::Ptr node) {
+            auto expr = to<mir::ConstantVectorExpr>(node);
+
+            for(auto &el : expr->vectorElements) {
+                vectorElements.push_back(el);
```

Comment:
Clone the expressions? I think we are performing a deep copy in FIR. Actually we should check what we do for other MIR nodes.

Reply:
Yeah i think you are right.